### ISSUE TITLE
Configuration -x flag issues

### ISSUE THREAD

**Issue by lolleko** [clangd/clangd#612]
I have a weird issue when trying to apply the `-x` flag via configuration.

Setup (simplified since I can't share the exact config):

```
If:
  PathMatch: .*\.h
CompileFlags:
  Add: -xobjective-c++-header
---
If:
  PathMatch: .*\.cpp
CompileFlags:
  Add: -xobjective-c++
```

```
[
	{
		"file": "source.cpp", // relies on some objective-c++ includes
		"command": "clang++  -ILOTSOFINCLUDES -std=c++14 source.cpp",
		"directory": "xyzrepo"
	},
]
```  
I can see that the flags are appended correctly int the output log:

```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ --driver-mode=g++ -ILOTSOFINCLUDES -std=c++14 source.cpp -xobjective-c++
```  
---

**However clangd does not appear to parse the file as objective-c++ I get a lot of objective c related errors.**

The same happens if I append `-xobjective-c++` manually to the compdb (with the config (.clangd) disabled):

```
[
	{
		"file": "source.cpp", // relies on some objective-c++ includes
                "command": "clang++  -ILOTSOFINCLUDES -std=c++14 source.cpp -xobjective-c++",
		"directory": "xyzrepo"
	},
]
```
Output:
```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ --driver-mode=g++ -ILOTSOFINCLUDES -std=c++14 source.cpp -xobjective-c++
```

---
But if I insert `-x-objective-c++` manually to the comb db **in front of the input file** it works fine and I get no errors:

```
[
	{
		"file": "source.cpp", // relies on some objective-c++ includes
                "command": "clang++  -ILOTSOFINCLUDES -std=c++14 -xobjective-c++ source.cpp",
		"directory": "xyzrepo"
	},
]
```
Output:
```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ --driver-mode=g++ -ILOTSOFINCLUDES -std=c++14 -xobjective-c++ source.cpp
```
---
**TL;DR** `-xobjective-c++` only seems to work correctly if inserted **in front** of the input file therefore this flag can't be added via a configuration  file

**Comment by gislan** [comment]
Thanks for the bug report.

This is a known issue. We are tracking it in #555
